audiomentations/augmentations/room_simulator.py

```diff
@@ -0,0 +1,355 @@
+            self.min_source_x, self.max_source_x
+        )
+        self.parameters["source_y"] = random.uniform(
+            self.min_source_y, self.max_source_y
+        )
+        self.parameters["source_z"] = random.uniform(
+            self.min_source_z, self.max_source_z
+        )
+
+        # Clamp between 0 and room dimensions
+        self.parameters["source_x"] = max(
+            0,
+            min(
+                self.parameters["size_x"] - self.padding,
+                self.parameters["source_x"],
+            ),
+        )
+        self.parameters["source_y"] = max(
+            0,
+            min(
+                self.parameters["size_y"] - self.padding,
+                self.parameters["source_y"],
+            ),
+        )
+        self.parameters["source_z"] = max(
+            0,
+            min(
+                self.parameters["size_z"] - self.padding,
+                self.parameters["source_z"],
+            ),
+        )
+
+        self.parameters["mic_radius"] = random.uniform(
+            self.min_mic_radius, self.max_mic_radius
+        )
+        self.parameters["mic_azimuth"] = random.uniform(
+            self.min_mic_azimuth, self.max_mic_azimuth
+        )
+        self.parameters["mic_elevation"] = random.uniform(
+            self.min_mic_elevation, self.max_mic_elevation
+        )
+
+        # Convert to cartesian coordinates according to ADM
+        mic_x = self.parameters["source_x"] - self.parameters["mic_radius"] * np.cos(
+            self.parameters["mic_elevation"]
+        ) * np.sin(self.parameters["mic_azimuth"])
+        mic_y = self.parameters["source_y"] + self.parameters["mic_radius"] * np.cos(
+            self.parameters["mic_elevation"]
+        ) * np.cos(self.parameters["mic_azimuth"])
+        mic_z = self.parameters["source_z"] + self.parameters["mic_radius"] * np.sin(
+            self.parameters["mic_elevation"]
+        )
+
+        # Clamp between 0 and room dimensions
+        self.parameters["mic_x"] = max(
+            0, min(self.parameters["size_x"] - self.padding, mic_x)
+        )
+        self.parameters["mic_y"] = max(
+            0, min(self.parameters["size_y"] - self.padding, mic_y)
+        )
+        self.parameters["mic_z"] = max(
+            0, min(self.parameters["size_z"] - self.padding, mic_z)
+        )
+
+    def apply(self, samples, sample_rate):
+        try:
+            import pyroomacoustics as pra
+        except ImportError:
+
+            print(
+                "Failed to import pyroomacoustics. Maybe it is not installed? "
+                "To install the optional pyroomacoustics dependency of audiomentations,"
+                " do `pip install audiomentations[extras]` or simply "
+                " `pip install pyroomacoustics`",
+                file=sys.stderr,
+            )
+            raise
+
+        assert samples.dtype == np.float32
+
+        # It makes no sense to use stereo data.
```

Comment:
Could you explain why?

Reply:
> Could you explain why?
 It needs some assumption as to how you map the stereo source in 3D space. In order to avoid making assumptions I just convert it to mono, however:

> What if we just convolve both channels with the same generated impulse response?

This is a good idea.